packages/compose/src/hooks/use-reduced-motion/index.js

```diff
@@ -10,13 +10,28 @@ import useMediaQuery from '../use-media-query';
  */
 const IS_IE = window.navigator.userAgent.indexOf( 'Trident' ) >= 0;
 
+/**
+ * Force reduced motion behavior at build time
+ *
+ * @type {boolean}
+ */
+let FORCE_REDUCED_MOTION = false;
+
+// Prefer the try/catch to allow bundlers to replace `process.env.FORCE_REDUCED_MOTION` completely.
+// Checks like `typeof process === 'object' && …` will be safe, but may short-circuit if a bundler
+// has replaced `process.env.FORCE_REDUCED_MOTION` with `true`, but `window.process`
+// remains undefined.
```

Comment:
That makes sense, but it also probably goes against the fact that bundlers try to also tree-shake these. For instance here, the try/catch might be kept (needs testing) while otherwise, the check is completely removed. Thoughts?

Reply:
A better approach may be to strip this entirely from the published package [as I mentioned below](https://github.com/WordPress/gutenberg/pull/16679#issuecomment-516081474), but I'm not sure.

Specifically on this point, here are my thoughts:

- Requiring package consumers to have a special build setup seems unreasonable, so unguarded `process.env.FORCE_REDUCED_MOTION` seems unreasonable.
- Ideally, this would be determined at build time, replaced by a constant, and dead code elimination could remove parts of the code. That would require a construct like `if ( process.env.FORCE_REDUCED_MOTION ) { /* block could be dead-code eliminated */ }`. That doesn't seem possible given the previous point.
- In order to provide safety for package consumers (guard the `process` usage) without requiring a special build setup, that would require something like:
  ```js
  if (
    typeof process === 'object' &&    // Check will remain at runtime
    process.env &&                    // Check will remain at runtime
    process.env.FORCE_REDUCED_MOTION  // Candidate for build-time replacement, may just be `true` or `false`
  ) {
    // Dead code elimination is very unlikely to occur here.
  }
  ```
  This seems to indicate that we can't have both _safety_ and _replacement with dead code elimination_. Additionally, depending on the build setup, it's entirely possible to have `process` be `undefined`, but `process.env.FORCE_REDUCED_MOTION` replaced with `true`, which would make this snippet completely broken.
- If we accept the above points, that seems to leave us with just the `try/catch` version. A consumer could opt-in to the `FORCE_REDUCED_MOTION`, consumers don't need any specific build/bundle setup, and there's only a minimal bundle/runtime penalty of a few bytes of JavaScript and a try/catch check on module evaluation. This seems to be the only reasonable approach to me, unless we can completely remove this from the published package.

Reply:
Thanks for the detailed explanation. That makes sense. The reason we have this env variable is that we want to disable animations in e2e tests. The ideal scenario would be to allow disabling animations at runtime. I didn't find a way to do it at that time.